pkg/common/cloudconfig/azure/azure_test.go

```diff
@@ -0,0 +1,260 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+// Package azure defines azure cloud provider configuration.
+package azure
+
+import (
+	"reflect"
+	"testing"
+
+	"sigs.k8s.io/cloud-provider-azure/pkg/azclient"
+)
+
+func TestLoadCloudConfigFromFile(t *testing.T) {
+	tests := map[string]struct {
+		filePath       string
+		expectErr      bool
+		expectedConfig *CloudConfig
+	}{
+		"file path is empty": {
+			filePath:  "",
+			expectErr: true,
+		},
+		"failed to open file": {
+			filePath:  "./test/not_exist.yaml",
+			expectErr: true,
+		},
+		"failed to unmarshal file": {
+			filePath:  "./test/azure_config_invalid.yaml",
+			expectErr: true,
+		},
+		"succeeded to load config": {
+			filePath: "./test/azure_config_valid.yaml",
+			expectedConfig: &CloudConfig{
+				ARMClientConfig: azclient.ARMClientConfig{
+					Cloud:     "AzurePublicCloud",
+					TenantID:  "00000000-0000-0000-0000-000000000000",
+					UserAgent: "fleet-net-controller-manager",
+				},
+				AzureAuthConfig: azclient.AzureAuthConfig{
+					UseManagedIdentityExtension: true,
+					UserAssignedIdentityID:      "",
+					AADClientID:                 "",
+					AADClientSecret:             "",
+				},
+				Location:       "eastus",
+				SubscriptionID: "00000000-0000-0000-0000-000000000000",
+				ResourceGroup:  "test-rg",
+			},
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			config, err := LoadCloudConfigFromFile(test.filePath)
+			if test.expectErr {
+				if err == nil {
+					t.Errorf("failed to test LoadCloudConfigFromFile: expected error, got nil")
+				}
+			} else {
+				if err != nil {
+					t.Errorf("failed to test LoadCloudConfigFromFile: expected no error, got %v", err)
+				}
+				if !reflect.DeepEqual(config, test.expectedConfig) {
+					t.Errorf("config parsed does not match, expected: %v, got: %v", test.expectedConfig, config)
+				}
+			}
```

Comment:
```suggestion
			if got := err != nil; got != test.wantErr {
				t.Fatalf("LoadCloudConfigFromFile() got error %v, wantErr %v", err, test.wantErr)
			}
			if cmp.Diff(config, test.expectedConfig) {
			      ....
			}
```
use diff is better as the structure contains many fields.

Reply:
fixed